BitFaster.Caching/Lazy/DisposableAtomic.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace BitFaster.Caching.Lazy
+{
+    // TODO: is this actually even needed? Or is the approach in ScopedAsyncAtomic sufficient? E.g. rely on IsValueCreated at dispose time, scoped owns tracking dispose and is already thread safe.
+    // requirements for IDisposable atomic
+    // if value !created, no dispose, cannot create - throws object disposed exception
+    // if created, dispose value
+    [DebuggerDisplay("IsValueCreated={IsValueCreated}, Value={ValueIfCreated}")]
+    public class DisposableAtomic<K, V> : IDisposable where V : IDisposable
```

Comment:
Could also create a ScopedLazy, allowing the user to substitute in Lazy behavior if preferred to Atomic.

This would work the same way - check the value is created property when disposing.